core/src/main/java/org/elasticsearch/cluster/metadata/IndexStateMetaData.java

```diff
@@ -0,0 +1,170 @@
+            throw new IllegalArgumentException("expected field name but got a " + parser.currentToken());
+        }
+        String currentFieldName = null;
+        String clusterUUID = null;
+        IndexMetaData indexMetaData = null;
+        do {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (INDEX_METADATA_FIELD.equals(currentFieldName)) {
+                    indexMetaData = IndexMetaData.Builder.fromXContent(parser);
+                } else {
+                    throw new IllegalArgumentException("Unexpected object [" + currentFieldName + "]");
+                }
+            } else if (token.isValue()) {
+                if (CLUSTER_UUID_FIELD.equals(currentFieldName)) {
+                    clusterUUID = parser.text();
+                } else {
+                    throw new IllegalArgumentException("Unexpected field [" + currentFieldName + "]");
+                }
+            } else {
+                throw new IllegalArgumentException("Unexpected token " + token);
+            }
+        } while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT);
+
+        if (indexMetaData == null) {
+            throw new ElasticsearchParseException("Could not parse indexMetaData from x-content");
+        }
+        if (clusterUUID == null) {
+            clusterUUID = IndexStateMetaData.CLUSTER_UUID_NA_VALUE;
+        }
+        return new IndexStateMetaData(indexMetaData, clusterUUID);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(CLUSTER_UUID_FIELD, getClusterUUID());
+        builder.startObject(INDEX_METADATA_FIELD);
+        IndexMetaData.Builder.toXContent(getIndexMetaData(), builder, params);
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * Test whether the current cluster UUID is the same as the given one.
+     */
+    public boolean isSameClusterUUID(final String otherClusterUUID) {
+        assert clusterUUID != null;
+        assert otherClusterUUID != null;
+        return clusterUUID.equals(otherClusterUUID);
+    }
+
+    @Override
+    public String toString() {
+        return "IndexStateMetaData[clusterUUID=" + clusterUUID + ", indexUUID=" + indexMetaData.getIndexUUID() +
+               ", indexName=" + indexMetaData.getIndex().getName() + ", state=" + indexMetaData.getState() +
+               ", noOfShards=" + indexMetaData.getNumberOfShards() + ", noOfReplicas=" + indexMetaData.getNumberOfReplicas() + "]";
+    }
+
+    private static final ToXContent.Params FORMAT_PARAMS = new MapParams(Collections.singletonMap("binary", "true"));
+
+    /**
+     * StateFormat that can read and write {@link IndexStateMetaData}
+     */
+    public static MetaDataStateFormat<IndexStateMetaData> FORMAT =
```

Comment:
we should remove `IndexMetaData#FORMAT`and `IndexMetaData#FORMAT_PARAMS` because they are now unused because of this